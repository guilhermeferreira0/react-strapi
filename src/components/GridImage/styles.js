import styled, { css } from 'styled-components';
import { Container as TextComponent } from '../TextComponent/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    & > ${TextComponent} {
      margin-bottom: ${theme.spacings.xhuge};
    }
  `}`;

export const Grid = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.spacings.large};
  `}`;

export const GridElement = styled.div`
  ${({ theme }) => css`
    overflow: hidden;
    border-radius: 8px;
  `}`;

export const Image = styled.img`
  ${({ theme }) => css`
    width: 100%;
    transition: all .5s ease-in-out;
    cursor: pointer;
    border-radius: 8px;

    &:hover {
      transform: scale(1.4) rotate(10deg);
    }
  `}`;
