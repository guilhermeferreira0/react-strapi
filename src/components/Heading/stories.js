import { Heading } from '.';
import { parameters } from '../../../.storybook/preview';

export default {
  title: 'Heading',
  component: Heading,
  args: {
    children: 'NÃ£o sei',
  },
  argTypes: {
    children: { type: 'string' },
  },
  parameters: {
    backgrounds: {
      default: 'dark',
    },
  },
};

export const Light = (args) => <Heading {...args} />;
export const Dark = (args) => <Heading {...args} colorDark={true} />;

Light.parameters = {
  backgrounds: {
    default: 'light',
  },
};

Light.args = {
  children: 'O texto esta claro',
  colorDark: false,
};
