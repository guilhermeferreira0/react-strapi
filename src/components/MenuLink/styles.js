import styled, { css } from 'styled-components';

export const Container = styled.a`
  ${({ theme }) => css`
    display: block;
    text-decoration: none;
    font-size: ${theme.fonts.sizes.small};
    padding: ${theme.spacings.small};
    color: ${theme.colors.primaryColor};
    position: relative;
    scale: 1;
    transition: scale 0.5s ease-in-out;


    &::after {
      content: '';
      position: absolute;
      bottom: 0.75rem;
      left: 50%;
      width: 0%;
      height: 0.2rem;
      background-color: ${theme.colors.secondaryColor};
      transition: all 0.5s ease-in-out;
    }

    &:hover::after {
      left: 25%;
      width: 50%;
    }

    &:hover {
      scale: 1.1;
    }

    @media ${theme.media.ltemedium} {
      &:hover::after {
      left: 20.5%;
      width: 50%;
    }
    }
  `}`;
