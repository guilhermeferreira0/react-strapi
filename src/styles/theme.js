export const theme = {
  colors: {
    primaryColor: 'lightgray',
    secondaryColor: '#dc143c',
    white: '#FFFFFF',
    lightBg: '#FFFFFF',
    darkBg: '#0A1128',
  },
  fonts: {
    family: {
      default: "'Open Sans', sans-serif",
      secondary: "'Montserrat', sans-serif",
    },
    sizes: {
      xsmall: '0.8rem',
      small: '1.6rem',
      medium: '2.4rem',
      large: '3.2rem',
      xlarge: '4rem',
      xxlarge: '4.8rem',
      huge: '5.6rem',
      xhuge: '6.2rem',
    },
  },
  spacings: {},
};
