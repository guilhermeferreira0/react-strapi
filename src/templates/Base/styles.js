import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    margin-top: 8%;
    width: 100%;
    height: 100%;
  `}`;
