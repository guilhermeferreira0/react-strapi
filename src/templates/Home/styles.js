import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  ${(props) => css`
    background-color: ${props.background};
  `}
`;
