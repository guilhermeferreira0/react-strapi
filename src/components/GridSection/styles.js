import styled, { css } from 'styled-components';
import { Title as HeadingContainer } from '../Heading/styles';
import { Container as TextComponent } from '../TextComponent/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    ${TextComponent} {
      margin-bottom: ${theme.spacings.xhuge};
    }
  `}`;

export const Grid = styled.div`
  ${({ theme }) => css`
    counter-reset: grid-counter;
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.spacings.large};

    @media ${theme.media.ltemedium} {
      grid-template-columns: 1fr;
    }
  `}`;

export const GridElement = styled.div`
  ${({ theme }) => css`
    ${HeadingContainer} {
      position: relative;
      left: 7rem;
    }

    ${HeadingContainer}::before {
      content: counter(grid-counter);
      counter-increment: grid-counter;
      position: absolute;
      font-size: ${theme.fonts.sizes.huge};
      top: -2rem;
      left: -7rem;
      transform: rotate(10deg);
    }
  `}`;
