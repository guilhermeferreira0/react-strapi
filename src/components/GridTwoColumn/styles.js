import styled, { css } from 'styled-components';
import { Title } from '../Heading/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: 1fr 2fr;
    align-items: center;
    gap: ${theme.spacings.large};

    @media ${theme.media.ltemedium} {
      grid-template-columns: 1fr;
      text-align: center;
    }

    ${Title} {
      margin-bottom: ${theme.spacings.xlarge};
    }
  `}`;

export const TextContainer = styled.div`
  ${({ theme }) => css`
    @media ${theme.media.ltemedium} {
      margin-bottom: ${theme.spacings.large};
    }
  `}`;

export const ImageContainer = styled.div`
  ${({ theme }) => css``}`;

export const Image = styled.div`
  ${({ theme }) => css``}`;
