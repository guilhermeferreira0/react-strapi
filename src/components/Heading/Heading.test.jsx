import { Heading } from '../Heading';
import { renderTheme } from '../../styles/renderTheme';
import { screen } from '@testing-library/react';
import { theme } from '../../styles/theme';

describe('<Heading /> ', () => {
  it('Should render with default values', () => {
    renderTheme(<Heading>Texto</Heading>);
    const heading = screen.getByRole('heading', { name: 'Texto' });

    expect(heading).toHaveStyle({
      color: 'black',
      'font-size': theme.fonts.sizes.xhuge,
      'text-transform': 'none',
    });
  });

  it('Should render with white color', () => {
    renderTheme(<Heading colorDark>Texto</Heading>);
    const heading = screen.getByRole('heading', { name: 'Texto' });

    expect(heading).toHaveStyle({
      color: theme.colors.white,
    });
  });

  it('Should render correct heading sizes', () => {
    renderTheme(<Heading size="small">Texto</Heading>);
    const heading = screen.getByRole('heading', { name: 'Texto' });

    expect(heading).toHaveStyle({
      'font-size': theme.fonts.sizes.medium,
    });
  });

  // it('Should render correct font-size when using mobile', () => {
  //   renderTheme(<Heading size="huge">Texto</Heading>);
  //   const heading = screen.getByRole('heading', { name: 'Texto' });

  //   expect(heading).toHaveStyleRule('font-size', theme.fonts.sizes.medium, {
  //     media: theme.media.ltemedium,
  //   });
  // });

  it('Should render with text uppercase', () => {
    renderTheme(<Heading uppercase>Texto</Heading>);
    const heading = screen.getByRole('heading', { name: 'Texto' });

    expect(heading).toHaveStyle({ 'text-transform': 'uppercase' });
  });

  it('Should render correct heading element', () => {
    const { container } = renderTheme(<Heading as="h6">Texto</Heading>);
    const heading = screen.getByRole('heading', { name: 'Texto' });
    const h6 = container.querySelector('h6');

    expect(h6.tagName.toLowerCase()).toBe('h6');
  });
});
