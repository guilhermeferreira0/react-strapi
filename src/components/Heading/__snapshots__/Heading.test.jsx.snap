// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Heading />  Should Match a snapshot 1`] = `
.c0 {
  color: black;
  font-size: 6.2rem;
  text-transform: none;
}

@media (max-width: 768px) {
  .c0 {
    font-size: 4rem;
  }
}

<h1
  class="c0"
>
  Texto
</h1>
`;
